src/Microsoft.ML.FastTree/Representation/TreeRegressorCollection.cs

```diff
@@ -0,0 +1,44 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+using Microsoft.ML.Trainers.FastTree.Internal;
+
+namespace Microsoft.ML.FastTree.Representation
+{
+    /// <summary>
+    /// A list of <see cref="TreeRegressor"/>. To compute the output value of a <see cref="TreeRegressorCollection"/>, we need to compute
+    /// the output values of all trees in <see cref="Trees"/>, scale those values via <see cref="TreeWeights"/>, and finally sum the scaled
+    /// values and <see cref="Bias"/> up.
+    /// </summary>
+    public class TreeRegressorCollection
```

Comment:
>TreeRegressorCollection [](start = 17, length = 23)

I do not like this calling something a collection if it does not implement `ICollection`. I am not BTW suggesting that you name it collection, since it differs from a collection in many important respects. Rather let's name it what it has always been named... `TreeEnsemble`. Was anything wrong with that name? Why are we changing it?

Reply:
It will be TreeEnsemble in iteration 21.

---
In reply to: [252499668](https://github.com/dotnet/machinelearning/pull/2243#discussion_r252499668) [](ancestors = 252499668)